app/routes/session.py

```diff
@@ -169,9 +174,27 @@ def _set_questionnaire_supplementary_data(
         survey_id=metadata["survey_id"],
         sds_dataset_id=new_sds_dataset_id,
     )
+    # ensure any required lists for the schema are included in the supplementary data
+    _validate_supplementary_data_lists(
+        supplementary_data=supplementary_data, schema=schema
+    )
     questionnaire_store.set_supplementary_data(supplementary_data["data"])
 
 
+def _validate_supplementary_data_lists(
```

Comment:
I think we should do the validation when `validate_supplementary_data_v1` runs. Thoughts?

Reply:
Yep 👍  - agree its better there, and makes the testing better too, moved it all over

Reply:
Now reverted as per other comment